### PR TITLE
Add description override codegenerator option

### PR DESCRIPTION
**Pull request by matthchr**
Use this option to override the Subnet IPConfigurations field to explain that it is removed if it crosses 2000 entries.

- [x] this PR contains documentation
- [ ] this PR contains tests
- [ ] this PR contains YAML Samples

### CODE REVIEW

File: v2/tools/generator/internal/codegen/code_generator.go [hunk 1]
```diff
@@ -199,6 +199,7 @@ func createAllPipelineStages(
 		pipeline.StripUnreferencedTypeDefinitions(),
 
 		pipeline.RenameProperties(configuration.ObjectModelConfiguration),
+		pipeline.OverrideDescriptions(configuration), // After flatten and rename to make easier to use
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/codegen/code_generator.go:202]: Suggest enforcing this with use of RequiresPrerequisiteStages() when defining the stage.

File: v2/tools/generator/internal/codegen/pipeline/override_descriptions.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+				newDef, err := visitor.VisitDefinition(def, def.Name())
+				if err != nil {
+					return nil, err
+				}
+
+				result.Add(newDef)
+			}
+
+			if err := configuration.ObjectModelConfiguration.Description.VerifyConsumed(); err != nil {
+				return nil, errors.Wrap(err, "verifying description augment")
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/codegen/pipeline/override_descriptions.go:38]:
```suggestion
				return nil, errors.Wrap(err, "verifying description augmentation")
```

**Review by theunrepentantgeek — APPROVED** [review]

**Comment by matthchr** [comment]
@microsoft-github-policy-service rerun

**Pull request merged**